src/qibolab/platform.py

```diff
@@ -45,17 +46,16 @@ def unroll_sequences(
     readout_map = defaultdict(list)
     channels = {pulse.channel for sequence in sequences for pulse in sequence}
     for sequence in sequences:
-        total_sequence.extend(sequence)
+        total_sequence += sequence
         # TODO: Fix unrolling results
         for pulse in sequence:
             if pulse.type is PulseType.READOUT:
                 readout_map[pulse.id].append(pulse.id)
 
         length = sequence.duration + relaxation_time
-        pulses_per_channel = sequence.pulses_per_channel
         for channel in channels:
-            delay = length - pulses_per_channel[channel].duration
-            total_sequence.append(Delay(duration=delay, channel=channel))
+            delay = length - sequence.channel_duration(channel)
+            total_sequence[channel].append(Delay(duration=delay, channel=channel))
```

Comment:
Now, we should start removing the channel from the `Pulse`-like:
```suggestion
            total_sequence[channel].append(Delay(duration=delay))
```